src/com/google/javascript/jscomp/ProductionCoverageInstrumentationCallback.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import static com.google.common.base.Strings.lenientFormat;
+
+import com.google.debugging.sourcemap.Base64VLQ;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+
+/**
+ * Instrument advanced coverage for javascript.
+ */
+final class ProductionCoverageInstrumentationCallback extends
+    NodeTraversal.AbstractPostOrderCallback {
+
+  private static final String INSTRUMENT_CODE_FUNCTION_NAME = "instrumentCode";
+  private static final String INSTRUMENT_CODE_FILE_NAME = "InstrumentCode.js";
+
+  /**
+   * The compiler runs an earlier pass that combines all modules and renames them appropriately.
+   * This constant represents what the INSTRUMENT_CODE_FILE_NAME module will be renamed to by the
+   * compiler and this will be used to make the correct call to INSTRUMENT_CODE_FUNCTION_NAME.
+   */
+  private static final String MODULE_RENAMING = "module$exports$instrument$code";
+
+  /**
+   * INSTRUMENT_CODE_FILE_NAME will contain an instance of the instrumentCode class and that
+   * instance name is stored in this constant.
+   */
+  private static final String INSTRUMENT_CODE_INSTANCE = "instrumentCodeInstance";
+
+  private final AbstractCompiler compiler;
+  private final ParameterMapping parameterMapping;
+  boolean visitedInstrumentCodeFile = false;
+  private String functionName = "Anonymous";
```

Comment:
Don't need this class member. It can be method-local.

Reply:
In this PR, the method can be made local I agree but it is class scope in anticipation of additional PRs that implement branch coverage. In that case, this field will be reused to avoid making redundant calls to `getBestLValueName`.